querydsl-jpa/src/main/java/com/querydsl/jpa/HibernateHandler.java

```diff
@@ -73,11 +74,11 @@ public boolean createNativeQueryTyped() {
     @SuppressWarnings("deprecation")
     @Override
     public boolean transform(Query query, FactoryExpression<?> projection) {
-        if (query instanceof NativeQuery) {
+        try {
             ResultTransformer transformer = new FactoryExpressionTransformer(projection);
-            query.unwrap(NativeQuery.class).setResultTransformer(transformer);
+            query.unwrap(org.hibernate.query.Query.class).setResultTransformer(transformer);
```

Comment:
Remove instance of checks,  just attempt to unwrap. This is because the query may be wrapped in various layers of proxies.

Also, we should unwrap a `Query` here and not a `NativeQuery` because this code is also used for regular queries.